contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1952,7 +1952,17 @@ pre_transform_target_entry(ResTarget *res, ParseState *pstate,
 				colname_start = pstate->p_sourcetext + res->location;
 				last_dot = colname_start;
 				while(*colname_start != '\0')
-				{	
+				{
+					/*
+					 * comment follow up with column like : 
+					 *
+					 * 'SELECT table1.c2--table1.REPGETTEXT('
+					 * 
+					 * will cause crash if we don't break the searching
+					 * for the last_dot position
+					 */
+					if (*colname_start == '-' && *(colname_start+1) == '-')
```

Comment:
1. colname_start + 1 may not be valid
2. we should check `open_square_bracket == 0 && double_quotes % 2 == 0` along with existing condition

Reply:
colname_start + 1 concern is fine, since the while condition is : 
```
*colname_start != '\0'
```

meaning the current colname_start is not '\0' and there's always a '\0' at the end of string, so colname_start +1 won't be invalid , at least it'll be '\0'